### PR TITLE
Folder modification

### PR DESCRIPTION
**Pull request by pillip8282**
modify folder structures for preventing the reverse reference between a framework folder and a netutils 
* changes
1) move tls from /os/net to /external
2) move modules in netuils from an apps folder to an external folder
3) remove the route table features come from uIP

### CODE REVIEW

File: apps/examples/artik_demo/artik_demo.c [hunk 1]
```diff
@@ -30,16 +30,16 @@
...
+#include <protocols/netlib.h>
+#include <protocols/webclient.h>
 
 #include <net/if.h>
 
 #include <dm/dm_error.h>
 #include <dm/dm_connectivity.h>
 
-#include <apps/netutils/wifi/slsi_wifi_api.h>
+#include <protocols/wifi/slsi_wifi_api.h>
```
Comment (davidfather) [inline apps/examples/artik_demo/artik_demo.c]: wifi is not a protocol, but a library.

File: external/Makefile [hunk 2]
```diff
@@ -97,6 +100,7 @@ include netlib/Make.defs
 include json/Make.defs
 include mbedtls/Make.defs
 
+
```
Comment (davidfather) [inline external/Makefile]: Please, remove this additional empty line.

File: external/codecs/Make.defs [hunk 3]
```diff
@@ -15,7 +15,7 @@
 # language governing permissions and limitations under the License.
 #
 ###########################################################################
-# apps/netutils/ftpc/Make.defs
+# apps/netutils/codecs/Make.defs
```
Comment (davidfather) [inline external/codecs/Make.defs]: apps/netutils/codecs/Make.defs
==>
external/codecs/Make.defs

File: external/dhcpd/Make.defs [hunk 4]
```diff
@@ -15,7 +15,7 @@
 # language governing permissions and limitations under the License.
 #
 ###########################################################################
-# apps/netutils/codecs/Make.defs
+# apps/netutils/dhcpd/Make.defs
```
Comment (davidfather) [inline external/dhcpd/Make.defs]: apps/netutils/dhcpd/Make.defs
==> 
external/protocols/dhcpd/Make.defs

File: external/dhcpd/Make.defs [hunk 5]
```diff
@@ -15,7 +15,7 @@
...
 # Adds selected applications to apps/ build
```
Comment (davidfather) [inline external/dhcpd/Make.defs]: Adds selected applications to apps/ build
==>
Adds selected applications to external/ build

File: external/codecs/Make.defs [hunk 6]
```diff
@@ -15,7 +15,7 @@
...
 # Adds selected applications to apps/ build
```
Comment (davidfather) [inline external/codecs/Make.defs]: Adds selected applications to apps/ build
==>
Adds selected applications to external/ build

File: external/libcoap/Make.defs [hunk 7]
```diff
@@ -51,6 +51,24 @@
...
+CSRCS += libcoap/str.c
+CSRCS += libcoap/subscribe.c
+CSRCS += libcoap/uri.c
+
+DEPPATH += --dep-path libcoap
+VPATH += :libcoap
+
 endif
 
+
```
Comment (davidfather) [inline external/libcoap/Make.defs]: Please, remove this empty line

File: external/webclient/webclient.c [hunk 8]
```diff
@@ -85,18 +85,18 @@
...
 
 #include "../webserver/http_string_util.h"
 #include "../webserver/http_client.h"
-#include <apps/netutils/webclient.h>
+#include <protocols/webclient.h>
 #if defined(CONFIG_NETUTILS_CODECS)
 #  if defined(CONFIG_CODECS_URLCODE)
 #    define WGET_USE_URLENCODE 1
-#    include <apps/netutils/urldecode.h>
+#    include <protocols/urldecode.h>
```
Comment (davidfather) [inline external/webclient/webclient.c]: urldecode is not a protocol.

File: external/webclient/webclient.c [hunk 9]
```diff
@@ -85,18 +85,18 @@
...
 #  endif
 #  if defined(CONFIG_CODECS_BASE64)
-#    include <apps/netutils/base64.h>
+#    include <protocols/base64.h>
```
Comment (davidfather) [inline external/webclient/webclient.c]: base64 is not a protocol

File: apps/examples/artik_demo/artik_demo.c [hunk 10]
```diff
@@ -30,16 +30,16 @@
 #include <tinyara/config.h>
 #include <apps/shell/tash.h>
 
-#include <apps/netutils/dhcpc.h>
-#include <apps/netutils/netlib.h>
-#include <apps/netutils/webclient.h>
+#include <protocols/dhcpc.h>
+#include <protocols/netlib.h>
```
Comment (davidfather) [inline apps/examples/artik_demo/artik_demo.c]: netlib is not a protocol.

Comment (pillip8282) [reply apps/examples/artik_demo/artik_demo.c]: done

File: apps/examples/artik_demo/artik_demo.c (hunk 1 above)
Comment (pillip8282) [reply apps/examples/artik_demo/artik_demo.c]: I will change it

File: external/Makefile (hunk 2 above)
Comment (pillip8282) [reply external/Makefile]: I removed the addtional line

File: external/codecs/Make.defs (hunk 3 above)
Comment (pillip8282) [reply external/codecs/Make.defs]: I changed all wrong file pathes in Make.defs in the external folder

**Pull request merged**